source/StepBro.Core/Parser/StepBroListener.ExpressionMethodCall.cs

```diff
@@ -746,9 +746,14 @@ private enum ArgumentInsertState
                 var p = parameters[i];
                 if (state == ArgumentInsertState.ExtensionInstance)
                 {
-                    suggestedAssignmentsOut.Add(new SBExpressionData(extensionInstance));
+                    SBExpressionData extensionInstanceData = new SBExpressionData(extensionInstance);
+                    suggestedAssignmentsOut.Add(extensionInstanceData);
                     thisParameterHandled = true;
                     state = ArgumentInsertState.Initial;
+                    if (IsParameterAssignableFromArgument(p, extensionInstanceData))
+                        matchScore += 5; // If we have the correct extension instance then it is more correct than a method that does not use the extension instance
```

Comment:
Only the decrease should be necessary.

Reply:
Incorrect, there exist "Contains" methods that do not have an ExtensionInstance that would then also have 1000 points like this would.